_posts/2018-06-11-2018-foreman-survey-analysis.md

```diff
@@ -0,0 +1,406 @@
+Handbook. That clearly needs another look in terms of promotion and possibly
+restructuring. I'm also concerned in the rise of Bug Reporting & Triage as a
+help category - that means (a) we've got a *lot* more bugs (which we also saw in
+the decrease in release stability rating above), or (b) it's not easy enough to
+submit bugs. I'm inclined to believe it's most (a), but I am planning an upgrade
+to Redmine soon, so perhaps we can improve the workflows there too.
+
+The developer ratings follow as similar pattern as the user ratings above -
+static or decreasing, with RFCs a particular sore point.
+
+## <a name="conclusions"></a> Conclusions
+
+I'll start by saying that these results don't surprise me. As the Community
+Lead, I get to chat with a lot of folks and hear about their problems and
+frustrations in private, so I was already aware that satisfaction was down. If
+anything, I'm happy it's not lower.
+
+It's also important to note that this is not a *bad* result in isolation - if I
+didn't show the 2017 data, it would look pretty good! This is not a cause for
+sadness, but it *is* a call to action if we're to reverse this trend in next
+year's survey.
+
+I've saved one graph for last - what the community thinks we should work on
+next. Let's take a look:
+
+![](/static/images/blog_images/2018-survey-images/next-1.svg)<!-- -->
+
+No surprise that bugs and stability top the list, given the rest of the data,
+but let's look at this with a different lens...
+
+Taken together, we have issues with bug counts, stability, new user experiences,
+upgrade processes, and release frequency. It's my belief that the solution to
+this is making Foreman run in a container. The implications of such a change are
+huge:
+
+* Having complete control of the runtime environment of Foreman will considerably reduce the matrix of support
+  * Today, every conversation with a user must start with "What OS are you on?"
+* Removes much of our packaging overhead
+* Simplfies plugins (probably, that needs some definition)
+* Removes (or at least largely simplifies) the need for our complex installer.
+
+This should lead to much greater stability, faster work on bugs (because the
+runtime is always the same), and a better experience for new-users (because
+there's no conflicts with other services on the same box, or weirdness in a
+particular OS). There's also some benefit from a developer perspective, as
+getting a dev environment up *could* be made easier.
+
+From this perspective, let's look at that graph again. 10% of the community want
+this feature directly (marked "Containers - Foreman"), but the argument can be
+made that such a move is beneficial for the Bugs, and support for host
+containers in Foreman itself (marked "Containers - Hosts"). One could also argue
+that's a step towards better HA support, since containers are far more flexible
+in that regard. Putting all that together, 67.37% of the
+community are served by such a move.
```

Comment:
I think that's a big stretch, just putting Foreman in a container(s) without fundamental changes to the application and plugins will not give anyone highly available Foreman.

If you want to leave it as is, which I don't mind if you do :)  please `s/community are served/community could be served/g`

Reply:
None of the container work will function without significant architecture changes anyway, but you are correct. I'll reword slightly